lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -166,6 +166,37 @@ def paths_to_3d_segments(paths, zs=0, zdir='z'):
         segments.append(path_to_3d_segment(path, pathz, zdir))
     return segments
 
+def path_to_3d_segment_with_codes(path, zs=0, zdir='z'):
+    '''Convert a path to a 3D segment with path codes.'''
+
+    if not iterable(zs):
+        zs = np.ones(len(path)) * zs
+
+    seg = []
+    codes = []
+    pathsegs = path.iter_segments(simplify=False, curves=False)
+    for (((x, y), code), z) in zip(pathsegs, zs):
```

Comment:
If you use `izip_longest` here, you don't need to allocate a large array of ones in `zs` first.